lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php

```diff
@@ -70,9 +70,9 @@ class ClassMetadataFactory extends AbstractClassMetadataFactory
     private $embeddablesActiveNesting = array();
 
     /**
-     * @param EntityManager $em
+     * @param EntityManagerInterface $em
      */
-    public function setEntityManager(EntityManager $em)
+    public function setEntityManager(EntityManagerInterface $em)
```

Comment:
This was a BC for me. I was overriding this method to get to the EM in metadata factory.